arc-mlir/src/include/arc/arc-ops.td

```diff
@@ -193,13 +193,24 @@ def MakeTuple
   let results = (outs AnyTuple);
 
   let verifier = [{
-    unsigned NumOperands = this->getOperation()->getNumOperands();
-    auto TupleTy =
-        this->getOperation()->getResult(0)->getType().dyn_cast<TupleType>();
+    auto Operation = this->getOperation();
+    auto OperandTys = Operation->getOperands().getTypes();
+    unsigned NumOperands = Operation->getNumOperands();
+    auto TupleTy = Operation->getResult(0).getType().dyn_cast<TupleType>();
```

Comment:
You want a `cast<>()` here, as if the tuple is not a tuple type you want the assert in debug mode. If it triggers something is very broken elsewhere.